HesoyamHospital/web-application-front/src/app/feedback/admin/surveys-results/surveys-results.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnInit } from '@angular/core';
+import { FeedbackService } from '../../services/feedback.service';
+import { SectionDTO } from '../../services/feedback.service';
+import {MeanDTO } from '../../services/feedback.service';
+import {Frequency } from '../../services/feedback.service';
+
+@Component({
+  selector: 'app-surveys-results',
+  templateUrl: './surveys-results.component.html',
+  styleUrls: ['./surveys-results.component.css']
+})
+export class SurveysResultsComponent implements OnInit {
+
+  displayedRows: ['How would you rate doctors proffesionalism? ', 'How would you rate doctors kindness?', 'How would you rate doctors expertise? ', 'In which measures does a doctor follows medical ethics principles?'];
+  displayedColumns2: string[] = ['questionOne', 'questionTwo', 'questionThree', 'questionFour', 'questionFive'];
+  displayedColumns: string[] = ['questionOne', 'questionTwo', 'questionThree', 'questionFour'];
+  public dataSource: SectionDTO[] = [];
+  public dataSourceStaff: SectionDTO[] = [];
+  public dataSourceHygiene: SectionDTO[] = [];
+  public dataSourceEquipment: SectionDTO[] = [];
+  public meanValuesPerQuestion: MeanDTO[] = [];
+  public meanValuesPerQuestionStaff: MeanDTO[] = [];
+  public meanValuesPerQuestionHygiene: MeanDTO[] = [];
+  public meanValuesPerQuestionEquipment: MeanDTO[] = [];
+  public meanValuesPerSection : MeanDTO[] = [];
+  public frequenciesDoctor: number[] = [];
+  public frequenciesStaff: number[] = [];
+  public frequenciesEquipment: number[] = [];
+  public frequenciesHygiene: number[] = [];
+
+  constructor(private _feedbackService: FeedbackService) { }
+
+  ngOnInit(): void {
+    this._feedbackService.getDoctorSections().subscribe((data) => {
+       this.dataSource = data;
+       
+          this._feedbackService.getStaffSections().subscribe((data) => {
```

Comment:
Why are all these methods nested within each other?
Is there any way to make this more understandable?